iocore/net/SSLNetVConnection.cc

```diff
@@ -508,6 +516,17 @@ SSLNetVConnection::net_read_io(NetHandler *nh, EThread *lthread)
           return;
         }
       }
+      // move over to the socket if we haven't already
+      if (this->handShakeBuffer) {
```

Comment:
Same comment here.  We only want to consider flipping back to the socket BIO if we still have the buffer support active.  So this is a narrower test than just checking that the context is client side.